File: suites/03_data_structures/comparisons/btree_operations.go
package main
import "fmt"
func branchWalk(depth, fanout int64) int64 {
	if depth == 0 { return fanout }
	var total int64
	for c := int64(0); c < fanout; c++ { total += branchWalk(depth-1, fanout-1) + c }
	return total
}
func main() { fmt.Println(branchWalk(5, 6)) }

File: suites/03_data_structures/comparisons/btree_operations.rs
fn branch_walk(depth: i64, fanout: i64) -> i64 {
    if depth == 0 { return fanout; }
    let mut total: i64 = 0;
    for c in 0..fanout { total += branch_walk(depth - 1, fanout - 1) + c; }
    total
}
fn main() { println!("{}", branch_walk(5, 6)); }
